complex/ft_msg.c

```diff
@@ -290,6 +309,48 @@ int ft_recv_dgram(void)
 	return -FI_ETIMEDOUT;
 }
 
+int ft_recv_dgram_flood(void)
+{
+	struct timespec s, e;
+	int credits, ret;
+	int64_t poll_time = 0;
+	int expected = ft.xfer_iter;
+	int got	     = 0;
+	do {
+		if (ft_rx.credits > (ft_rx.max_credits >> 1)) {
+			ret = ft_post_recv_bufs();
+			if (ret)
+				return ret;
+		}
+
+		credits = ft_rx.credits;
+
+		ret = ft_comp_rx();
+		if (credits != ft_rx.credits) {
+			int num_creds = ft_rx.credits - credits;
+			poll_time=0;
+			got+=num_creds;
+		}
+		if (ret)
+			return ret;
+
+		if(got >= expected) {
+			return 0;
```

Comment:
I think we're okay on the ack packet.  The sender doesn't receive too many packets, so hopefully this one doesn't get lost.  If it does, the sender should just eventually time out.

For early termination, I think this will work.  After the receiver sees the done message, it simply continues to pull all completions off the CQ and count them as received.  Once the CQ is empty, it's really done and can ack the sender.